rcl_logging_interface/src/logging_dir.c

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <rcutils/allocator.h>
+#include <rcutils/error_handling.h>
+#include <rcutils/filesystem.h>
+#include <rcutils/find.h>
+#include <rcutils/get_env.h>
+#include <rcutils/repl_str.h>
+#include <rcutils/strdup.h>
+
+#include "rcl_logging_interface/rcl_logging_interface.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
```

Comment:
nit: this isn't needed here.

This is only needed in pure C include files (`.h`), because they can be included from a CPP file.
In the case of a C source file, the build system picks the right compiler based on the extension.